docs/admin/scale/index.md

```diff
@@ -0,0 +1,69 @@
+
+## General concepts
+
+Coder runs workspace operations in a queue. The number of concurrent builds will be limited to the number of provisioner daemons across all coderd replicas.
+
+- **coderd**: Coder’s primary service. Learn more about [Coder’s architecture](../../about/architecture.md)
+- **coderd replicas**: Replicas (often via Kubernetes) for high availability, this is an [enterprise feature](../../enterprise.md)
+- **concurrent workspace builds**: Workspace operations (e.g. create/stop/delete/apply) across all users
+- **concurrent connections**: Any connection to a workspace (e.g. SSH, web terminal, `coder_app`)
+- **provisioner daemons**: Coder runs one workspace build per provisioner daemon. One coderd replica can host many daemons
+- **scaletest**: Our scale-testing utility, built into the `coder` command line.
+
+```text
+2 coderd replicas * 30 provisioner daemons = 60 max concurrent workspace builds
+```
+
+## Infrastructure recommendations
+
+### Concurrent workspace builds
+
+Workspace builds are CPU-intensive, as it relies on Terraform. Various [Terraform providers](https://registry.terraform.io/browse/providers) have different resource requirements. When tested with our [kubernetes](https://github.com/coder/coder/tree/main/examples/templates/kubernetes) template, `coderd` will consume roughly 8 cores per 30 concurrent workspace builds. For effective provisioning, our helm chart prefers to schedule [one coderd replica per-node](https://github.com/coder/coder/blob/main/helm/values.yaml#L110-L121).
+
+To support 120 concurrent workspace builds, for example:
+
+- Create a cluster/nodepool with four 8-core nodes (AWS: `t3.2xlarge` GCP: `e2-highcpu-8`)
+- Run coderd with 4 replicas, 30 provisioner daemons each. (`CODER_PROVISIONER_DAEMONS=30`)
+- Ensure Coder's [PostgreSQL server](../../admin/configure.md#postgresql-database) can use up to 1.5 cores
+
+## Recent scale tests
+
+| Environment        | Users | Concurrent builds | Concurrent connections (SSH) | Concurrent connections (web) | Last tested  |
+| ------------------ | ----- | ----------------- | ---------------------------- | ---------------------------- | ------------ |
+| Kubernetes (GKE)   | 1200  | 120               | 10,000                       | 10,000                       | Jan 10, 2022 |
+| Docker (Single VM) | 500   | 50                | 10,000                       | 10,000                       | Dec 20, 2022 |
+
+## Scale testing utility
+
+Since Coder's performance is highly dependent on the templates and workflows you support, we recommend using our scale testing utility against your own environments.
+
+The following command will run our scale test against your own Coder deployment. You can also specify a template name and any parameter values.
+
+```sh
+coder scaletest create-workspaces \
+    --count 1000 \
+    --template "kubernetes" \
+    --concurrency 0 \
```

Comment:
You may want to explain here what `concurrency 0` and `cleanup-concurrency 0` mean.

Reply:
@deansheather what do you think about making `0` the default value for these? I was a bit confused why this defaults to 1.

`concurrency 0` means the client will request 1000 workspaces at once (no limit). `concurrency 200` means that the client will request a max of 200 requests at a given time.

Reply:
Sure. We could make it a required flag so admins consciously opt-in to infinite rather than infinite by default.